test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -7034,6 +7035,402 @@ public async Task TestPacMan_PreviewInstallPackage_BuildIntegrated_NullVersion_T
+                {
+                    Id = contosoPackageIdentity.Id,
+                    Version = "1.0.0"
+                };
+                ContosoReal.AddFile("lib/net461/contosoA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    privateRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ContosoReal);
+
+                primarySources.Add(primarySource);
+                secondarySources.AddRange(primarySources);
+                secondarySources.Add(new PackageSource(privateRepositoryPath, "PrivateRepository"));
+
+                SourceRepositoryProvider primarySourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(primarySources);
+                SourceRepositoryProvider secondarySourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(secondarySources);
+
+                ISettings testSettings = Settings.LoadSpecificSettings(testSolutionManager.SolutionDirectory, "NuGet.Config");
+                var token = CancellationToken.None;
+                var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                var nuGetPackageManager = new NuGetPackageManager(
+                    secondarySourceRepositoryProvider,
+                    testSettings,
+                    testSolutionManager,
+                    deleteOnRestartManager);
+
+                var msBuildNuGetProject = testSolutionManager.AddNewMSBuildProject();
+                var msBuildNuGetProjectSystem = msBuildNuGetProject.ProjectSystem as TestMSBuildNuGetProjectSystem;
+                var packagesConfigPath = msBuildNuGetProject.PackagesConfigNuGetProject.FullPath;
+
+                // Pre-Assert
+                // Check that the packages.config file does not exist
+                Assert.False(File.Exists(packagesConfigPath));
+                // Check that there are no packages returned by PackagesConfigProject
+                var packagesInPackagesConfig = (await msBuildNuGetProject.PackagesConfigNuGetProject.GetInstalledPackagesAsync(token)).ToList();
+                Assert.Equal(0, packagesInPackagesConfig.Count);
+                Assert.Equal(0, msBuildNuGetProjectSystem.References.Count);
+
+                // Act and Assert
+                var ex = await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+                {
+                    await nuGetPackageManager.PreviewInstallPackageAsync(msBuildNuGetProject, contosoPackageIdentity.Id,
+                    new ResolutionContext(), new TestNuGetProjectContext(), primarySourceRepositoryProvider.GetRepositories(), secondarySourceRepositoryProvider.GetRepositories(), token);
+                });
+
+                // Even though Contoso.A 1.0.0 exist in ExternalRepository it wouldn't be picked for install.
+                // PrivateRepository is passed in secondary sources but for packages.config project not considered.
```

Comment:
There are scenarios where secondary sources are considered correct/ 

Can we add tests for that?

Reply:
@nkolev92 
I was not able to find other scenarios where secondary sources are considered. Please note in above test `secondary sources` is not incorrect, simply not searched(considered) from it. 
Could you be able to point what are other possible scenarios where secondary sources are considered?
I already checked update/downgrade action but it's actually just `uninstall(no source is considered)` + `install(above tested case)` action.

Reply:
My recollection is that primary sources are considered for direct dependencies, while both primary and secondary are considered for transitive dependencies.

Reply:
Ok. I added 2 more tests.

https://github.com/NuGet/NuGet.Client/blob/cd77b927c2e12d315113898ba8dc05801af48d60/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7324-L7325

https://github.com/NuGet/NuGet.Client/blob/cd77b927c2e12d315113898ba8dc05801af48d60/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs#L7451-L7452